benchmarktests/target_secret_mysql.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package benchmarktests
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/vault/api"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+// Constants for test
+const (
+	MySQLSecretTestType   = "mysql_secret"
+	MySQLSecretTestMethod = "GET"
+	MySQLUsernameEnvVar   = VaultBenchmarkEnvVarPrefix + "MYSQL_USERNAME"
+	MySQLPasswordEnvVar   = VaultBenchmarkEnvVarPrefix + "MYSQL_PASSWORD"
+)
+
+func init() {
+	// "Register" this test to the main test registry
+	TestList[MySQLSecretTestType] = func() BenchmarkBuilder { return &MySQLSecret{} }
+}
+
+// Postgres Secret Test Struct
+type MySQLSecret struct {
+	pathPrefix string
+	roleName   string
+	header     http.Header
+	config     *MySQLSecretTestConfig
+	logger     hclog.Logger
+}
+
+// Main Config Struct
+type MySQLSecretTestConfig struct {
+	MySQLDBConfig   *MySQLDBConfig   `hcl:"db_connection,block"`
+	MySQLRoleConfig *MySQLRoleConfig `hcl:"role,block"`
+}
+
+// MySQL DB Config
+type MySQLDBConfig struct {
+	Name                   string   `hcl:"name,optional"`
+	PluginName             string   `hcl:"plugin_name,optional"`
+	PluginVersion          string   `hcl:"plugin_version,optional"`
+	VerifyConnection       *bool    `hcl:"verify_connection,optional"`
+	AllowedRoles           []string `hcl:"allowed_roles,optional"`
+	RootRotationStatements []string `hcl:"root_rotation_statements,optional"`
+	PasswordPolicy         string   `hcl:"password_policy,optional"`
+	ConnectionURL          string   `hcl:"connection_url"`
+	Username               string   `hcl:"username,optional"`
+	Password               string   `hcl:"password,optional"`
+	DisableEscaping        bool     `hcl:"disable_escaping,optional"`
+	MaxOpenConnections     int      `hcl:"max_open_connections,optional"`
+	MaxIdleConnections     int      `hcl:"max_idle_connections,optional"`
+	MaxConnectionLifetime  string   `hcl:"max_connection_lifetime,optional"`
+	UsernameTemplate       string   `hcl:"username_template,optional"`
+	TLSCertificateKey      string   `hcl:"tls_certificate_key,optional"`
+	TLSCACertificate       string   `hcl:"tls_ca_certificate,optional"`
+	TLSServerName          string   `hcl:"tls_server_name,optional"`
+	TLSSkipVerify          bool     `hcl:"tls_skip_verify,optional"`
+}
+
+// MySQL Role Config
+type MySQLRoleConfig struct {
+	Name                 string `hcl:"name,optional"`
+	DBName               string `hcl:"db_name,optional"`
+	DefaultTTL           string `hcl:"default_ttl,optional"`
+	MaxTTL               string `hcl:"max_ttl,optional"`
+	CreationStatements   string `hcl:"creation_statements"`
+	RevocationStatements string `hcl:"revocation_statements,optional"`
```

Comment:
in the docs this is a slice

Reply:
Ill leave it as a string and update the docs.  Good catch